[Regression] GLES3 shaders error with: undefined variable "uv"

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
<!-- Specify commit hash if non-official. -->
2042d62 -> 2042d6214e1ab302a1d9a0a4bd8a018bf6ee679b

**OS/device including version:**
<!-- Specify GPU model and drivers if graphics-related. -->
Ubuntu 18.04 64bit
NVidia 390.77

**Issue description:**
<!-- What happened, and what was expected. -->
Countless errors with shaders.
e.g.
`
ERROR: _display_error_with_code: CanvasShaderGLES3: Vertex Program Compilation Failed:
0(178) : error C1008: undefined variable "uv"
0(178) : error C1008: undefined variable "uv"
0(179) : error C1008: undefined variable "uv"
0(179) : error C1008: undefined variable "uv"

   At: drivers/gles3/shader_gles3.cpp:173.
ERROR: get_current_version: Method/Function Failed, returning: __null
   At: drivers/gles3/shader_gles3.cpp:351.
ERROR: bind: Condition ' !version ' is true. returned: false
   At: drivers/gles3/shader_gles3.cpp:123.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
`
**Steps to reproduce:**
Still trying to figure out how to reproduce this cause it happened to my project but I cannot isolate it to share.

**Minimal reproduction project:**
<!-- Recommended as it greatly speeds up debugging. Drag and drop a zip archive to upload it. -->

BTW, you should have a dump of the shader code before this error:
```
ERROR: _display_error_with_code: CanvasShaderGLES3: Vertex Program Compilation Failed:
0(178) : error C1008: undefined variable "uv"
```
Could you post it here?

From the terminal
<details>
1: #version 330
2: #define GLES_OVER_GL
3: #define USE_TEXTURE_RECT
4: #define USE_MATERIAL
5: precision highp float;
6: precision highp int;
7: 
8: layout(location = 0) in highp vec2 vertex;
9: /* clang-format on */
10: layout(location = 3) in vec4 color_attrib;
11: 
12: #ifdef USE_SKELETON
13: layout(location = 6) in uvec4 bone_indices; // attrib:6
14: layout(location = 7) in vec4 bone_weights; // attrib:7
15: #endif
16: 
17: #ifdef USE_TEXTURE_RECT
18: 
19: uniform vec4 dst_rect;
20: uniform vec4 src_rect;
21: 
22: #else
23: 
24: #ifdef USE_INSTANCING
25: 
26: layout(location = 8) in highp vec4 instance_xform0;
27: layout(location = 9) in highp vec4 instance_xform1;
28: layout(location = 10) in highp vec4 instance_xform2;
29: layout(location = 11) in lowp vec4 instance_color;
30: 
31: #ifdef USE_INSTANCE_CUSTOM
32: layout(location = 12) in highp vec4 instance_custom_data;
33: #endif
34: 
35: #endif
36: 
37: layout(location = 4) in highp vec2 uv_attrib;
38: 
39: // skeleton
40: #endif
41: 
42: uniform highp vec2 color_texpixel_size;
43: 
44: layout(std140) uniform CanvasItemData { //ubo:0
45: 
46: 	highp mat4 projection_matrix;
47: 	highp float time;
48: };
49: 
50: uniform highp mat4 modelview_matrix;
51: uniform highp mat4 extra_matrix;
52: 
53: out highp vec2 uv_interp;
54: out mediump vec4 color_interp;
55: 
56: #ifdef USE_NINEPATCH
57: 
58: out highp vec2 pixel_size_interp;
59: #endif
60: 
61: #ifdef USE_SKELETON
62: uniform mediump sampler2D skeleton_texture; // texunit:-1
63: uniform highp mat4 skeleton_transform;
64: uniform highp mat4 skeleton_transform_inverse;
65: #endif
66: 
67: #ifdef USE_LIGHTING
68: 
69: layout(std140) uniform LightData { //ubo:1
70: 
71: 	// light matrices
72: 	highp mat4 light_matrix;
73: 	highp mat4 light_local_matrix;
74: 	highp mat4 shadow_matrix;
75: 	highp vec4 light_color;
76: 	highp vec4 light_shadow_color;
77: 	highp vec2 light_pos;
78: 	highp float shadowpixel_size;
79: 	highp float shadow_gradient;
80: 	highp float light_height;
81: 	highp float light_outside_alpha;
82: 	highp float shadow_distance_mult;
83: };
84: 
85: out vec4 light_uv_interp;
86: out vec2 transformed_light_uv;
87: 
88: out vec4 local_rot;
89: 
90: #ifdef USE_SHADOWS
91: out highp vec2 pos;
92: #endif
93: 
94: const bool at_light_pass = true;
95: #else
96: const bool at_light_pass = false;
97: #endif
98: 
99: #if defined(USE_MATERIAL)
100: 
101: /* clang-format off */
102: layout(std140) uniform UniformData { //ubo:2
103: int m_particles_anim_h_frames;
104: int m_particles_anim_v_frames;
105: bool m_particles_anim_loop;
106: 
107: 
108: };
109: /* clang-format on */
110: 
111: #endif
112: 
113: /* clang-format off */
114: 
115: 
116: /* clang-format on */
117: 
118: void main() {
119: 
120: 	vec4 color = color_attrib;
121: 
122: #ifdef USE_INSTANCING
123: 	mat4 extra_matrix_instance = extra_matrix * transpose(mat4(instance_xform0, instance_xform1, instance_xform2, vec4(0.0, 0.0, 0.0, 1.0)));
124: 	color *= instance_color;
125: 	vec4 instance_custom = instance_custom_data;
126: 
127: #else
128: 	mat4 extra_matrix_instance = extra_matrix;
129: 	vec4 instance_custom = vec4(0.0);
130: #endif
131: 
132: #ifdef USE_TEXTURE_RECT
133: 
134: 	if (dst_rect.z < 0.0) { // Transpose is encoded as negative dst_rect.z
135: 		uv_interp = src_rect.xy + abs(src_rect.zw) * vertex.yx;
136: 	} else {
137: 		uv_interp = src_rect.xy + abs(src_rect.zw) * vertex;
138: 	}
139: 	highp vec4 outvec = vec4(dst_rect.xy + abs(dst_rect.zw) * mix(vertex, vec2(1.0, 1.0) - vertex, lessThan(src_rect.zw, vec2(0.0, 0.0))), 0.0, 1.0);
140: 
141: #else
142: 	uv_interp = uv_attrib;
143: 	highp vec4 outvec = vec4(vertex, 0.0, 1.0);
144: #endif
145: 
146: #ifdef USE_PARTICLES
147: 	//scale by texture size
148: 	outvec.xy /= color_texpixel_size;
149: #endif
150: 
151: #define extra_matrix extra_matrix_instance
152: 
153: 	{
154: 		/* clang-format off */
155: 	{
156: 		float m_h_frames=float(m_particles_anim_h_frames);
157: 		float m_v_frames=float(m_particles_anim_v_frames);
158: 		outvec.xy.xy/=vec2(m_h_frames, m_v_frames);
159: 		int m_total_frames=(m_particles_anim_h_frames*m_particles_anim_v_frames);
160: 		int m_frame=int((float(m_total_frames)*instance_custom.z));
161: 		if (m_particles_anim_loop)
162: 		{
163: 					{
164: 			m_frame=(abs(m_frame)%m_total_frames);
165: 		}
166: ;
167: 		}
168: 		else
169: 		{
170: 					{
171: 			m_frame=clamp(m_frame, 0, (m_total_frames-1));
172: 		}
173: ;
174: 		}
175: 		float m_frame_w=(1.0/m_h_frames);
176: 		float m_frame_h=(1.0/m_v_frames);
177: 		uv.x=((uv.x*m_frame_w)+(m_frame_w*float((m_frame%m_particles_anim_h_frames))));
178: 		uv.y=((uv.y*m_frame_h)+(m_frame_h*float((m_frame/m_particles_anim_h_frames))));
179: 	}
180: 
181: 
182: 		/* clang-format on */
183: 	}
184: 
185: #ifdef USE_NINEPATCH
186: 
187: 	pixel_size_interp = abs(dst_rect.zw) * vertex;
188: #endif
189: 
190: #if !defined(SKIP_TRANSFORM_USED)
191: 	outvec = extra_matrix * outvec;
192: 	outvec = modelview_matrix * outvec;
193: #endif
194: 
195: #undef extra_matrix
196: 
197: 	color_interp = color;
198: 
199: #ifdef USE_PIXEL_SNAP
200: 	outvec.xy = floor(outvec + 0.5).xy;
201: #endif
202: 
203: #ifdef USE_SKELETON
204: 
205: 	if (bone_weights != vec4(0.0)) { //must be a valid bone
206: 		//skeleton transform
207: 
208: 		ivec4 bone_indicesi = ivec4(bone_indices);
209: 
210: 		ivec2 tex_ofs = ivec2(bone_indicesi.x % 256, (bone_indicesi.x / 256) * 2);
211: 
212: 		highp mat2x4 m;
213: 		m = mat2x4(
214: 					texelFetch(skeleton_texture, tex_ofs, 0),
215: 					texelFetch(skeleton_texture, tex_ofs + ivec2(0, 1), 0)) *
216: 			bone_weights.x;
217: 
218: 		tex_ofs = ivec2(bone_indicesi.y % 256, (bone_indicesi.y / 256) * 2);
219: 
220: 		m += mat2x4(
221: 					 texelFetch(skeleton_texture, tex_ofs, 0),
222: 					 texelFetch(skeleton_texture, tex_ofs + ivec2(0, 1), 0)) *
223: 			 bone_weights.y;
224: 
225: 		tex_ofs = ivec2(bone_indicesi.z % 256, (bone_indicesi.z / 256) * 2);
226: 
227: 		m += mat2x4(
228: 					 texelFetch(skeleton_texture, tex_ofs, 0),
229: 					 texelFetch(skeleton_texture, tex_ofs + ivec2(0, 1), 0)) *
230: 			 bone_weights.z;
231: 
232: 		tex_ofs = ivec2(bone_indicesi.w % 256, (bone_indicesi.w / 256) * 2);
233: 
234: 		m += mat2x4(
235: 					 texelFetch(skeleton_texture, tex_ofs, 0),
236: 					 texelFetch(skeleton_texture, tex_ofs + ivec2(0, 1), 0)) *
237: 			 bone_weights.w;
238: 
239: 		mat4 bone_matrix = skeleton_transform * transpose(mat4(m[0], m[1], vec4(0.0, 0.0, 1.0, 0.0), vec4(0.0, 0.0, 0.0, 1.0))) * skeleton_transform_inverse;
240: 
241: 		outvec = bone_matrix * outvec;
242: 	}
243: 
244: #endif
245: 
246: 	gl_Position = projection_matrix * outvec;
247: 
248: #ifdef USE_LIGHTING
249: 
250: 	light_uv_interp.xy = (light_matrix * outvec).xy;
251: 	light_uv_interp.zw = (light_local_matrix * outvec).xy;
252: 
253: 	mat3 inverse_light_matrix = mat3(inverse(light_matrix));
254: 	inverse_light_matrix[0] = normalize(inverse_light_matrix[0]);
255: 	inverse_light_matrix[1] = normalize(inverse_light_matrix[1]);
256: 	inverse_light_matrix[2] = normalize(inverse_light_matrix[2]);
257: 	transformed_light_uv = (inverse_light_matrix * vec3(light_uv_interp.zw, 0.0)).xy; //for normal mapping
258: 
259: #ifdef USE_SHADOWS
260: 	pos = outvec.xy;
261: #endif
262: 
263: 	local_rot.xy = normalize((modelview_matrix * (extra_matrix_instance * vec4(1.0, 0.0, 0.0, 0.0))).xy);
264: 	local_rot.zw = normalize((modelview_matrix * (extra_matrix_instance * vec4(0.0, 1.0, 0.0, 0.0))).xy);
265: #ifdef USE_TEXTURE_RECT
266: 	local_rot.xy *= sign(src_rect.z);
267: 	local_rot.zw *= sign(src_rect.w);
268: #endif
269: 
270: #endif
271: }
272: 
273: /* clang-format off */
274: 
ERROR: _display_error_with_code: CanvasShaderGLES3: Vertex Program Compilation Failed:
0(177) : error C1008: undefined variable "uv"
0(177) : error C1008: undefined variable "uv"
0(178) : error C1008: undefined variable "uv"
0(178) : error C1008: undefined variable "uv"

   At: drivers/gles3/shader_gles3.cpp:173.
ERROR: get_current_version: Method/Function Failed, returning: __null
   At: drivers/gles3/shader_gles3.cpp:351.
ERROR: bind: Condition ' !version ' is true. returned: false
   At: drivers/gles3/shader_gles3.cpp:123.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:380.
</details>

--- a/drivers/gles3/shaders/canvas.glsl
+++ b/drivers/gles3/shaders/canvas.glsl
@@ -145,6 +145,8 @@ void main() {
 
 #define extra_matrix extra_matrix_instance
 
+	//for compatibility with the fragment shader we need to use uv here
+	vec2 uv = uv_interp;
 	{
 		/* clang-format off */
 
@@ -152,6 +154,8 @@ VERTEX_SHADER_CODE
 
 		/* clang-format on */
 	}
+
+	uv_interp = uv;
 
 #ifdef USE_NINEPATCH
 

